### PR TITLE
Build & push to Docker Hub

### PR DESCRIPTION
**Pull request by minhducsun2002**
Docker Hub is so slooooow.

---
This change is [Reviewable](https://reviewable.io/reviews/natsukagami/kjudge/32)

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+on: push
+jobs:
+  build:
+    name: Publish to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Publish
+        uses: whoan/docker-build-with-cache-action@v3.1.1
```
Comment (natsukagami) [inline .github/workflows/docker.yml:11]: Why not docker's [own action](https://github.com/docker/build-push-action)?

Comment (minhducsun2002) [reply .github/workflows/docker.yml:11]: Never knew this existed. I will make changes then.

Comment (minhducsun2002) [reply .github/workflows/docker.yml]: Done.

**Comment by minhducsun2002** [comment]
Temporarily closing for testing. I'll reopen upon finishing.

**Pull request closed without merging**